File: Game/assets/scripts/chopper.lua
Chopper = setmetatable({}, { __index = LuieScript })
Chopper.__index = Chopper

function Chopper:new(entity)
    local obj = LuieScript.new(self, entity) -- Call base constructor
    return obj
end

function Chopper:OnStart()
    self.moveSpeed = 100
    print("Chopper position: x = " .. self.transform.position.x .. " y = " .. self.transform.position.y)
    self.sprite = GetSpriteComponent(self.entity)
    self.sprite.Sprite = "chopper-spritesheet"
    -- print(self.spriteComponent.Sprite)
end

function Chopper:OnUpdate(deltaTime)
    local xVel, yVel = self:GetEntityVelocity()
    if IsKeyRepeated(Keys.A) then
        xVel = xVel - (self.moveSpeed * deltaTime)
        self.sprite.SrcRect.y = self.sprite.Height * 3
    end
    if IsKeyRepeated(Keys.D) then
        xVel = xVel + (self.moveSpeed * deltaTime)
        self.sprite.SrcRect.y = self.sprite.Height * 1
    end
    if IsKeyRepeated(Keys.W) then
        yVel = yVel - (self.moveSpeed * deltaTime)
        self.sprite.SrcRect.y = self.sprite.Height * 0
    end
    if IsKeyRepeated(Keys.S) then
        yVel = yVel + (self.moveSpeed * deltaTime)
        self.sprite.SrcRect.y = self.sprite.Height * 2
    end

    if IsKeyPressed(Keys.Space) then
        print("Space was pressed")
    end

    if IsKeyReleased(Keys.Space) then
        print("Space was Released")
    end

    self:SetEntityVelocity(xVel, yVel)
end

File: Game/TankGame/Chopper.cs
using MeanScriptEngine;
using MeanScriptEngine.Components;
using MeanScriptEngine.Input;

public class Chopper : MeanScriptBehaviour
{
    private RigidBodyComponent _rigidBodyComponent;
    private AnimationComponent _animationComponent;
    private SpriteComponent _spriteComponent;

    private float _movespeed = 100;
    public override void OnStart()
    {
        _rigidBodyComponent = AddComponent<RigidBodyComponent>();
        _rigidBodyComponent.Velocity = new Vector2(500, 0);

        _animationComponent = GetComponent<AnimationComponent>();
        _spriteComponent = GetComponent<SpriteComponent>();

        _animationComponent.NumOfFrames = 2;
    }

    public override void OnUpdate(float deltaTime)
    {
        if (Transform.Position.X >= 600)
        {
            _rigidBodyComponent.Velocity.X = -50 * (_movespeed * deltaTime);
        }
        else if (Transform.Position.X <= 100)
        {
            _rigidBodyComponent.Velocity.X = 50 * _movespeed * deltaTime;
        }

        if (Input.IsKeyPressed(Key.P))
        {
            Log.Error("P is pressed");
            _animationComponent.FrameRateSpeed = 100;
            // _movespeed += 100;
        }
        else if (Input.IsKeyReleased(Key.P))
        {
            Log.Error("P is released");
            _animationComponent.FrameRateSpeed = 10;
        }
    }
}